src/meta-srv/src/procedure/create_table.rs

```diff
@@ -99,17 +113,126 @@ impl CreateTableProcedure {
             return Ok(Status::Done);
         }
 
-        self.creator.data.state = CreateTableState::DatanodeCreateTable;
+        self.creator.data.state = if expr.engine == MITO2_ENGINE {
+            CreateTableState::DatanodeCreateRegions
+        } else {
+            CreateTableState::DatanodeCreateTable
+        };
 
         Ok(Status::executing(true))
     }
 
-    async fn on_create_metadata(&self) -> Result<Status> {
-        let _timer = common_telemetry::timer!(
-            crate::metrics::METRIC_META_CREATE_TABLE_PROCEDURE_CREATE_META
-        );
+    fn create_region_request_template(&self) -> Result<PbCreateRegionRequest> {
+        let create_table_expr = &self.creator.data.task.create_table;
+
+        let column_defs = create_table_expr
+            .column_defs
+            .iter()
+            .enumerate()
+            .map(|(i, c)| {
+                let semantic_type = if create_table_expr.time_index == c.name {
+                    SemanticType::Timestamp
+                } else if create_table_expr.primary_keys.contains(&c.name) {
+                    SemanticType::Tag
+                } else {
+                    SemanticType::Field
+                };
+
+                ColumnDef {
+                    name: c.name.clone(),
+                    column_id: i as u32,
+                    datatype: c.datatype,
+                    is_nullable: c.is_nullable,
+                    default_constraint: c.default_constraint.clone(),
+                    semantic_type: semantic_type as i32,
+                }
+            })
+            .collect::<Vec<_>>();
+
+        let primary_key = create_table_expr
+            .primary_keys
+            .iter()
+            .map(|key| {
+                column_defs
+                    .iter()
+                    .enumerate()
+                    .find_map(|(i, c)| if &c.name == key { Some(i as u32) } else { None })
```

Comment:
I remember that elements in `primary_key` field in `CreateTableRequest` are the column ids. @waynexia 

If so, we should update the document in the proto file.
```suggestion
                    .find_map(|c| if &c.name == key { Some(c.column_id) } else { None })
```

Reply:
it's true